### ISSUE TITLE
Single-day events show up as two days

### ISSUE THREAD

**Issue by onitake** [KrissN/akonadi-ews#12]
When creating a new all-day event in Outlook for a single day, it shows up as occupying two days in KOrganizer.

This seems like a bug in akonadi-ews, perhaps because of a misinterpretation of the 00:00-00:00 timespan?

**Comment by KrissN** [comment]
In the current implementation akonadi-ews doesn't do much interpretation of the event retrieved from Exchange. The server provides the event in form of a iCalendar entry, which is sent to Akonadi for parsing. I don't remember right now, but I think the iCalendar format has a separate entry for "all-day event".

As mentioned in the 0.8.0 release announcement there is a plan to add write support for the calendar. This will likely end-up in a rewrite of the current implementation and move from retrieving the iCalendar entry to retrieving raw data and recreating the event from scratch. I will make sure to test all-day events to make sure that they work properly.

**Comment by onitake** [comment]
Awesome, thanks!

**Comment by KrissN** [comment]
Could this also happen because of this bug: https://bugs.kde.org/show_bug.cgi?id=362921 ?

**Comment by onitake** [comment]
I don't think it does, as I can happily create a single-day event on a local calendar and it will show up on that day only. Creating a spanning event (i.e. from 20:00 on day X to 8:00 on day X+1) works fine too.

This only happens when a single-day event is loaded from Exchange. Evolution and Outlook display the event just fine.

**Comment by KrissN** [comment]
Okay, thanks for the update.
